cmd/clusterctl/validation/validate_pods_test.go

```diff
@@ -0,0 +1,168 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"bytes"
+	"testing"
+
+	"golang.org/x/net/context"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1/testutil"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+func getPodWithContainerSpec(podName, namespace, containerName, containerImage string) corev1.Pod {
```

Comment:
nit: don't start the function name with "get"; this should be `podWithContainerSpec(...)`

Reply:
Fixed here for `podWithStatus ` method:
https://github.com/kubernetes-sigs/cluster-api/pull/679/files#diff-a363c04a3cb096347d5c77d779ec333fR27

I have deleted `podWithContainerSpec` method since it's not required anymore